iree/compiler/Dialect/LinalgExt/Transforms/Loops.cpp

```diff
@@ -0,0 +1,166 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtDialect.h"
+#include "iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "iree/compiler/Dialect/LinalgExt/Transforms/PassDetail.h"
+#include "iree/compiler/Dialect/LinalgExt/Transforms/Passes.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/SCF.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace mlir {
+namespace iree_compiler {
+namespace linalg_ext {
+namespace {
+
+struct BubbleSortConversion : public OpRewritePattern<linalg_ext::SortOp> {
+  using OpRewritePattern<linalg_ext::SortOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(linalg_ext::SortOp op,
+                                PatternRewriter& rewriter) const final {
+    if (!op.hasBufferSemantics()) return failure();
+    // This can be relaxed if the inputs and the outputs are not tied. Will need
+    // a copy op.
+    if (!llvm::all_of(llvm::zip(op.getInputOperands(), op.getOutputOperands()),
```

Comment:
Interesting requirement. This seems off, but I see why you have this here. Might have to revisit the `sort` op specification and how the `tensor` and `memref` versions are described. For now this is OK.

Reply:
As offline discussion, the op should not take inputs. The op definition is updated, so we no longer need the check.